### PR TITLE
Do not wait indefinitely for the API to reply

### PR DESCRIPTION
**Pull request by kubrickfr**
This PR implements recommendations & sane defaults documented in
http://docs.python-requests.org/en/master/user/advanced/#timeouts

it needs to be applied after #27

### CODE REVIEW

File: src/onelogin/api/client.py [hunk 1]
```diff
@@ -48,13 +48,22 @@ class OneLoginClient(object):
 
     CUSTOM_USER_AGENT = "onelogin-python-sdk %s" % __version__
 
-    def __init__(self, client_id, client_secret, region='us', max_results=1000):
+    def __init__(self, client_id, client_secret, region='us', max_results=1000, default_timeout=(3.05, 27)):
```
Comment (pitbulk) [inline src/onelogin/api/client.py]: Can you set the default value to None?

**Comment by pitbulk** [comment]
Please set None as default value and document this parameter on the README.

**Comment by kubrickfr** [comment]
The problem with `None` is that despite being default in `requests`, their documentation states that this is not a sane default.
Why do you want it set to None? If it's not to change the existing behaviour, I consider that the existing behaviour is buggy: connections should time-out at some point.

**Comment by kubrickfr** [comment]
I agree it's a breaking change and I think the options are:

1. Setting it to `None`, which makes the change non-breaking
2. Setting it to the value recommended by `requests`, which is `(3.05, 27)` (this PR) which should be _better_ for everybody except the users you're talking about, which I expect should be very few
3. Setting it to an arbitrarily high value
4. Setting it to a value that matches what onelogin does on their servers (you should be able to get this information), this way it should be non-breaking in the sense that it would still fail without this change, but silently, which is not good.

Personally, I think my preferred options are 2 and 4. 3 is based on guesswork and 1 is just not improving the situation for existing users.

**Comment by kubrickfr** [comment]
> @kubrickfr I see your point, but what happens if current customers have a slow net where 3.05 and 27 are low values? The script will then fail when currently works.

To clarify, I think it's wrong to say "when currently works". At the moment, the script will wait indefinitely but the server (onelogin's) will not.

**Comment by pitbulk** [comment]
Ok, then can you change it to 10, 60? I believe we are safe with those values.

**Comment by kubrickfr** [comment]
:+1: from what I can see on my side, the onelogin API timeout is indeed 60s + TCP handshake

so I'm cool with the values you suggest

**Pull request merged**